src/compiler/codegen/index.js

```diff
@@ -499,7 +499,7 @@ function genProps (props: Array<{ name: string, value: any }>): string {
 
 /* istanbul ignore next */
 function generateValue (value) {
-  if (typeof value === 'string') {
+  if (isString(value)) {
```

Comment:
What do you think about `return isString(value) ? transformSpecialNewlines(value) : JSON.stringify(value)` notation ?